src/Microsoft.AspNet.Mvc.Formatters.Json/Logging/JsonInputFormatterLoggerExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Mvc.Formatters.Json.Logging
+{
+    internal static class JsonInputFormatterLoggerExtensions
+    {
+        private static readonly Action<ILogger, string, Exception> _jsonInputFormatterCrashed;
+
+        static JsonInputFormatterLoggerExtensions()
+        {
+            _jsonInputFormatterCrashed = LoggerMessage.Define<string>(
+                LogLevel.Verbose,
+                1,
+                "Json input formatting failed, exception was '{0}'.");
```

Comment:
Well, for one, `Json` needs to be `JSON`. But aside from that, when logging an exception does there need to be a `{0}` in the message or will the system automatically append the exception details to the log anyway? I forget how the logging exception formatting stuff works.

But back to the error message itself, how about: `The JSON input formatter threw the following exception: {0}` (and if the `{0}` ends up not being needed, say `... threw an exception.`)